R/addRank.R

```diff
@@ -0,0 +1,48 @@
+#' Rank the sample information in a `SummarizedExperiment` Object
+#'
+#' The information stored in `colData` in \linkS4class{SummarizedExperiment}
+#' object can be ranked by `addRank` function.
+#'
+#' @param x \linkS4class{SummarizedExperiment} object
+#' @param ... Allow new parameters to be defined for this function.
+#' @param na.last  NA values are put last when \code{TRUE} or put first when
+#' \code{FALSE} or they are kept with rank NA when \code{"keep"}
+#' @param ties.method a character string indicating the method used in ranking
+#'
+#' @return a list shows the ranked components of
+#' \linkS4class{SummarizedExperiment} object
+#'
+#' @examples
+#' library(SummarizedExperiment)
+#' data(airway, package="airway")
+#' se <- airway
+#'
+#' addRank(se, na.last ="TRUE", ties.method = "last")
+#'
+#' @importFrom SummarizedExperiment SummarizedExperiment
+#'
+#' @export
+setGeneric("addRank", signature = "x",
+        function(x,...)
+            standardGeneric("addRank"))
+
+.is_integer <- function(t){
+    i <- seq_along(t)
+    for (i in t){
+        if(is.integer(t[[i]]) == FALSE){
+        t[-i]
+        }
+    }
+}
+
+setMethod("addRank", signature(x = "SummarizedExperiment"),
+            function(x,
+                    na.last = "TRUE",
+                    ties.method = c("average", "first", "last", "random", "max",
+                        "min"), ...){
+            col_data <- colData(x)
+            list <- col_data@listData
+            integerlist <- .is_integer(list) #lapply(list, .is_integer)
```

Comment:
Where does this come from? Are you here trying to test the integer status for multiple different coldata fields?

Reply:
I was trying to do it specific to integers, since it can be ranked correctly in every case. Because if there is something like sampleID for example, the ranking is not necessarily successful because sampleID is nominal. For instance sample1234 is not greater than sample1235 but in ranking it will show like it is. I might be thinking wrong, so in case that's what we want to get anyway , I can do it

Reply:
The idea is to get ranking only for a single selected field that the user would name as an argument. For instance:

`addRank(se, field="time", rank_field="time_rank")`


This would add to `se` a new field `time_rank` and that would contain the rank for the field `time`.

If user wants to do this for different fields, they can do it themselves in they way they need,.

Reply:
Thank you for the help! I guess I made it work this time but I am not sure about the function arguments part, especially if there is a need to keep "..."

Reply:
Should work so that if you call 

`addRank(se, field="time", rank_field="time_rank", na.last=TRUE, ties.method="first")`

then if can use all that and the `na.last` and `ties.method` can be passed to `rank` through the "..." mechanism. 

You could ask tvborman for help if not clear otherwise.